File: LeetCode/560. Subarray Sum Equals K/solution.py
from typing import List
from collections import defaultdict


class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        dic = defaultdict(int)
        dic[0] = 1

        ans = 0
        sum = 0

        for n in nums:
            sum += n
            if dic[sum - k]:
                ans += dic[sum - k]
            dic[sum] += 1

        return ans

File: LeetCode/560. Subarray Sum Equals K/solution.swift
class Solution {
    func subarraySum(_ nums: [Int], _ k: Int) -> Int {
        var prefixSumCounts: [Int: Int] = [0: 1]
        var currentPrefixSum = 0
        var totalCount = 0
        
        for num in nums {
            currentPrefixSum += num
            
            if let count = prefixSumCounts[currentPrefixSum - k] {
                totalCount += count
            }
            
            prefixSumCounts[currentPrefixSum, default: 0] += 1
        }
        
        return totalCount
    }
}
